src/connector.ts

```diff
@@ -26,13 +26,13 @@ export class Connector {
 	private _observers: Array<Observer> = []
 	private coreHandler: CoreHandler
 	private _config: Config
-	private _logger: Winston.LoggerInstance
+	private _logger: Logger
 	private _process: Process
 	private _settings?: INewsDeviceSettings
 	private _debug: boolean
 
-	constructor(logger: Winston.LoggerInstance, config: Config, debug: boolean) {
-		this._logger = logger
+	constructor(logger: Logger, config: Config, debug: boolean) {
+		this._logger = logger.tag('Connector')
```

Comment:
Just a thought: could we use `logger.tag(this.constructor.name)` or something similar that programatically finds the name of the class, instead of writing them as string literals every time?

Reply:
Yes we can. I'll have a look at it 😛